packages/react/src/Card/Card.tsx

```diff
@@ -86,11 +86,11 @@ const CardRoot = forwardRef<HTMLDivElement, CardProps>(
       if (React.isValidElement(child) && typeof child.type !== 'string') {
         if (
           isFragment(child) ||
-          child.type === CardImage ||
-          child.type === CardIcon ||
-          child.type === CardLabel ||
-          child.type === CardHeading ||
-          child.type === CardDescription
+          (child as React.ReactElement).type === CardImage ||
+          (child as React.ReactElement).type === CardIcon ||
+          (child as React.ReactElement).type === CardLabel ||
+          (child as React.ReactElement).type === CardHeading ||
+          (child as React.ReactElement).type === CardDescription
         ) {
           return true
         }
```

Comment:
Not one for this PR, but since we have this kind of check all over the codebase we might benefit from a small centralised [type predicate](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) which will remove the need for dozens of these type assertion throughout the codebase.

```tsx
// utils.ts
export const isValidReactElement = (element: unknown): element is React.ReactElement => {
  return React.isValidElement(child) && typeof child.type !== 'string');
};
```

```suggestion
     // Use the type predicate here
     if (isValidReactElement(child)) {
        if (
          isFragment(child) ||
          // So that TypeScript knows that `child` is a ReactElement here
          child.type === CardImage ||
          child.type === CardIcon ||
          child.type === CardLabel ||
          child.type === CardHeading ||
          child.type === CardDescription
        ) {
          return true
        }
```